cluster/mesos/cluster.go

```diff
@@ -168,12 +170,20 @@ func NewCluster(scheduler *scheduler.Scheduler, TLSConfig *tls.Config, master st
 
 // Handle callbacks for the events
 func (c *Cluster) Handle(e *cluster.Event) error {
+	// publish event to the watchQueue for external subscribers
+	c.watchQueue.Publish(e)
```

Comment:
Why don't we adapt this to the eventHandlers list? I am not sure that making this dependent on the watch queue makes sense.